src/common/headers.rs

```diff
@@ -94,6 +98,8 @@ pub struct Headers {
     /// `Accept` header might be used by HTTP clients to enforce server responses with content
     /// formatted in a specific way.
     accept: MediaType,
+    /// Hashmap reserved for storing custom headers.
+    custom_map: HashMap<String, String>,
```

Comment:
I thought about that too but then realised the struct is `Headers` so in practice it would look like:
```rust
let headers = Headers::new()
let bla_header = headers.custom_headers.get("bla");
```
or
```rust
let bla_header = headers.custom_map.get("bla");
```

maybe `custom_entries`?

I'm fine with any of them any way.

Reply:
Having something like
```rust
let bla_header = headers.custom_headers.get("bla");
``` 
looked a bit odd to me, this is why I chose to name the HashMap `custom_map`. 

However, `custom_entries` sounds better and makes more sense, so I think I will stick to this name.